### ISSUE TITLE
PowerTunnel sometimes not working after Android 12

### ISSUE THREAD

**Issue by Hans5958** [krlvm/PowerTunnel-Android#148]
Upon updating to Android 12, PowerTunnel started to be unstable. Last time on 11, PowerTunnel never broke, but upon updating to 12, it only works sometimes, especially to ISP blocked sites (Reddit in this case). Sometimes it works perfectly, no bugs, sometimes I couldn't access ANY sites, sometimes it just passess through without unblocking.

Some observed behaviors, keep in mind these are quite inconsistent:
1. No PowerTunnel: Firefox works except Reddit, Joey for Reddit doesn't work, other app works 
2. Clean state/app reset: Same as (1) ([Log file](https://github.com/krlvm/PowerTunnel-Android/files/11503999/Untitled.log.txt))
3. plus Strict DNS Mode: Firefox doesn't work at all, apps work including Joey for Reddit.
4. minus DNS module: Same as (1)
5. minus DNS module, plus Strict DNS Mode: Same as (1)

There is a similar issue on #140 but I don't think it is the same, so I put it on a separate issue.

I'm running Android 12 with One UI 4.1 in my Samsung Galaxy A31.

**Comment by krlvm** [comment]
Does it work in proxy mode or if you disable DNS plugin?
You can also test if [version 1.9.1](https://github.com/krlvm/PowerTunnel-Android/releases/tag/v1.9.1) works for you.

**Comment by Hans5958** [comment]
Still not working when disabling the DNS plugin (keep in mind Reddit is blocked here, so probably no surprise).

I may try the proxy mode or version 1.9.1 later.

PS. I edited the original post. I wrote that on the mobile earlier and now edited it on the computer.

**Comment by Hans5958** [comment]
I've tried 1.9.1 and the behaviour is the same as on 2.x.

**Comment by Hans5958** [comment]
Proxy also does not work, by the way.

I'm happy to fulfill specific log request, by the way.

**Comment by krlvm** [comment]
I don't think there's something helpful in logs, I will try to test it on Android 12 later

**Comment by Hans5958** [comment]
As I had nothing to do, I decided to dig deeper by trial and error. Hope this can be found useful. I tried my best to clear the data to solve cache issues. 

For information regarding the blocking in Indonesia, my country, see https://github.com/bebasid/KominFudge.

### Accessing Reddit using the official Reddit app (2023.31.0)

- ❌ Reddit is blocked
- ✔️ Reddit is accessible; not blocked

| ⬇️ | Mobile (3, [AS45727](https://bgp.tools/as/45727)) | Wi-Fi (IndiHome, [AS7713](https://bgp.tools/as/7713)) |
| - | - | - |
| *none* | ❌ | ❌ 
| PowerTunnel (2.7.3) | ❌ | ❌ 
| PowerTunnel (2.7.3) with DNS | ❌ | ✔️ 
| Nebulo (2.0-beta) | ❌ | ✔️ 
| 1.1.1.1 app (6.26) | ✔️ | ✔️ 
| 1.1.1.1 app (6.26) with WARP | ✔️ | ✔️ 
| Proton VPN (4.6.12.1) | ✔️ | ✔️ 

### Ads on Notebloc (4.3.7)

- ❌ Ads are not blocked
- ✔️ Ads are blocked

| ⬇️ | Mobile (3, [AS45727](https://bgp.tools/as/45727)) | Wi-Fi (IndiHome, [AS7713](https://bgp.tools/as/7713)) |
| - | - | - |
| *none* | ❌ | ❌ 
| PowerTunnel (2.7.3) with AdBlock enabled | ✔️ | ✔️
| Nebulo (2.0-beta) | ✔️ | ❌

Specification:
- Phone: Samsung Galaxy A31
	- OS: Android 12 with One UI 4.1
	- Android security patch level: 1 February 2023
- PowerTunnel: App data cleared every run
	- DNS Profile: Cloudflare 
- 1.1.1.1: App data cleared every run
- Reddit: App data cleared every run
- Notebloc: App data not cleared for convenience
- Proton VPN: App data not cleared for convenience
- Nebulo: App data not cleared for convenience
   - Server configuration: 1dot1dot1dot1.cloudflare-dns.com (DoT)
   - DNS rules: [StevenBlack/hosts](https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts)

**Comment by krlvm** [comment]
Do you have custom DNS enabled in PowerTunnel?

**Comment by Hans5958** [comment]
I cleared the app data and it is chosen as "Custom" and "Not set" by default.

**Comment by krlvm** [comment]
Try other options, such as Google or Cloudflare, I mean

**Comment by Hans5958** [comment]
Same as Nebulo. Updated the table.
